buildbot-conf/cordova.conf

```diff
@@ -0,0 +1,326 @@
+    return DescribedStep(ShellCommand, workdir=workdir, timeout=timeout, **kwargs)
+
+def NPM(npm_command, command=list(), what='code', **kwargs):
+    return SH(command=['npm', npm_command] + command, description='npm ' + npm_command + 'ing ' + what, **kwargs)
+
+def NPMInstall(command=list(), **kwargs):
+    # NOTE:
+    #      adding the --cache parameter so that we don't use the global
+    #      npm cache, which is shared with other processes
+    return NPM('install', command=command + [I('--cache=%(prop:workdir)s/npm_cache')], **kwargs)
+
+def NPMTest(**kwargs):
+    return NPM('test', **kwargs)
+
+def BuildbotClone(repourl, what='code', workdir=None, **kwargs):
+    if workdir is None:
+        workdir = what
+    return DescribedStep(Git, 'cloning ' + what, repourl=repourl, workdir=workdir, mode='full', method='clobber', shallow=True, **kwargs)
+
+def CordovaClone(project_name, **kwargs):
+    branch   = repo_branch_from_name(project_name)
+    repo_url = repo_url_from_name(project_name)
+    return BuildbotClone(repourl=repo_url, branch=branch, what=project_name, **kwargs)
+
+def MedicClone(project_name=None, **kwargs):
+    """
+    Clone repositories using medic's checkout.js script.
+    """
+
+    command = ['node', 'cordova-medic/bin/checkout.js', '--config=cordova-medic/cordova-repos.json']
+
+    if project_name is not None:
+        command    += ['--project=' + project_name]
+        description = 'cloning ' + project_name
+    else:
+        description = 'cloning all cordova projects'
+
+    return SH(command=command, description=description)
+
+def RMRF(path, description=None, **kwargs):
+
+    if description is None:
+        description = 'removing ' + path
+
+    # this is better than 'rm -rf' because it removes excessively long paths on Windows
+    js_script = "var s = require('shelljs'); console.log('removing {path}'); s.rm('-rf', '{path}');".format(path=path)
+
+    return SH(command=['node', '-e', js_script], description=description, **kwargs)
+
+def Set(name, value, **kwargs):
+    return DescribedStep(SetProperty, 'setting ' + name, property=name, value=value, **kwargs)
+
+def Download(mastersrc, slavedest):
+    return FileDownload(mastersrc=mastersrc, slavedest=slavedest, workdir=BASE_WORKDIR)
+
+####### SLAVES
+
+OSX_SLAVES     = ['cordova-ios-slave']
+WINDOWS_SLAVES = ['cordova-windows-slave']
+
+####### CHANGESOURCES
+
+# None, because Apache manages them, and we should not touch them.
+
+####### CHANGE FILTERS
+
+# None
+
+####### BUILDERS
+
+common_setup_steps = [
+
+    # set build properties
+    Set('repositoryname', render_repo_name),
+
+    # kill running emulators
+    SH(
+        command         = render_task_kill_command,
+        doStepIf        = can_find_running_tasks,
+        description     = 'killing running tasks',
+        haltOnFailure   = False,
+        flunkOnWarnings = False,
+        warnOnWarnings  = False,
+        decodeRC        = ALWAYS_SUCCESS,
+    ),
```

Comment:
I think killing all emulators and reinitializing them on each build is time consuming. Also it could lead to timeout build failures. I propose to remove this step.